src/System.Device.Gpio/build/net5.0/System.Device.Gpio.targets

```diff
@@ -0,0 +1,7 @@
+<Project>
+  <Target Name="EnsureProjectIsTargetingWindowsPlatformIfRequired"
+          BeforeTargets="Build"
+          Condition="'$(TargetFramework)' == 'net5.0' And $(RuntimeIdentifier.StartsWith('win'))">
```

Comment:
You only want to cause runnable applications to fail, not libraries.  It's ok for another library to target just `net5.0`.  Can you make sure this check only applies to apps?  Also it's best practice to have a way to disable such errors.  Consider adding a property to suppress this.

Reply:
Good point about not generating the error for libraries.  Relatedly, you probably want to use the `buildTransitive` folder instead of the `build` folder so that if an app transitively references this package it will get the build error.

Reply:
Also per offline conversation you might be able to use `AppHostRuntimeIdentifier` to catch the case where someone is building a framework-dependent app on Windows.

Reply:
If I do that, then folks creating framework-dependent apps on Windows with the intention of running them on Linux will get the error, which I actually believe will be one of the most common scenarios. Would it be ok to not protect against that case and instead let the runtime exception catch such scenarios given they should, in theory, be the least common case?

Reply:
> creating framework-dependent apps on Windows with the intention of running them on Linux

We could teach these folks to use `AppHostRuntimeIdentifier` but maybe that's overkill.  I'm fine with whatever you think is best here.  We could even forgo the whole targets if we don't think it'll help enough to warrant the added complexity.

Reply:
I think that as long as it catches some of the cases it is worth keeping, as a build time error is always preferred over runtime error, but just want to make sure that we don't catch false-positives by making our safety net too general.